api/src/components/bookmark/service.ts

```diff
@@ -1,92 +1,151 @@
 import * as client from 'lib/client';
+import * as I from 'interface';
 
-export const create = async (publicationId: string, userId: string) => {
-    const create = await client.prisma.publicationBookmarks.create({
+export const create = async (type: I.BookmarkType, entityId: string, userId: string) => {
+    const create = await client.prisma.bookmark.create({
         data: {
-            publicationId,
+            type,
+            entityId,
             userId
         },
         select: {
             id: true,
-            publicationId: true,
+            type: true,
+            entityId: true,
             userId: true
         }
     });
 
     return create;
 };
 
-export const get = async (publicationId: string, userId: string) => {
-    const bookmark = await client.prisma.publicationBookmarks.findFirst({
+export const get = async (id: string) => {
+    const bookmark = await client.prisma.bookmark.findUnique({
         where: {
-            publicationId,
+            id
+        },
+        select: {
+            id: true,
+            type: true,
+            entityId: true,
+            userId: true
+        }
+    });
+
+    return bookmark;
+};
+
+export const getByFields = async (type: I.BookmarkType, entityId: string, userId: string) => {
+    const bookmark = await client.prisma.bookmark.findFirst({
+        where: {
+            type,
+            entityId,
             userId
         },
         select: {
             id: true,
-            publicationId: true,
+            type: true,
+            entityId: true,
             userId: true
         }
     });
 
     return bookmark;
 };
 
-export const remove = async (bookmarkId: string) => {
-    const bookmark = await client.prisma.publicationBookmarks.delete({
+export const deleteBookmark = async (id: string) => {
+    const bookmark = await client.prisma.bookmark.delete({
         where: {
-            id: bookmarkId
+            id
         }
     });
 
     return bookmark;
 };
 
-export const getAll = async (userId: string) => {
-    const bookmark = await client.prisma.publicationBookmarks.findMany({
+export const getAll = async (userId: string, type: I.BookmarkType): Promise<I.PopulatedBookmark[]> => {
```

Comment:
we can use this service to filter bookmarks based on our needs, that means it could take an argument of type BookmarkWhereInput and then, from the 'getAll' controller we can provide {userId, type, entityId} as filters. This way, we can check if a specific bookmark exists, by fetching it like this: /bookmarks?type=PUBLICATION&entityId=pub_id_123. This will return an array with a single record if the bookmark exists or an empty array otherwise.

Reply:
Makes sense, I've changed it to allow an entityId filter too